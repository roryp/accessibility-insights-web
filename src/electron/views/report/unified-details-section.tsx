// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { ScanMetadata } from 'common/types/store-data/unified-data-interface';
import {
    makeDetailsSectionFC,
    ScanDetailInfo,
} from 'reports/components/report-sections/make-details-section-fc';

export function createDeviceNameItemInfo(scanMetadata: ScanMetadata): ScanDetailInfo {
    return {
        label: 'connected device name:',
        content: `${scanMetadata.deviceName} - ${scanMetadata.targetAppInfo.name}`,
    };
}

export const UnifiedDetailsSection = makeDetailsSectionFC(createDeviceNameItemInfo);
