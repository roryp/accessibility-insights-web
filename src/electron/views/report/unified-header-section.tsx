// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { androidAppTitle } from 'content/strings/application';
import { BrandWhite } from 'icons/brand/white/brand-white';
import * as React from 'react';
import * as styles from 'reports/components/report-sections/header-section.scss';

export const UnifiedHeaderSection = NamedFC('UnifiedHeaderSection', () => {
    return (
        <header>
            <div className={styles.reportHeaderBar}>
                <BrandWhite />
                <div className={styles.headerText}>{androidAppTitle}</div>
            </div>
        </header>
    );
});
