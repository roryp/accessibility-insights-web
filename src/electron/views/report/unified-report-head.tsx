// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { androidAppTitle } from 'content/strings/application';
import * as React from 'react';
import * as reportStyles from 'reports/automated-checks-report.styles';

import * as bundledStyles from '../bundled-renderer-styles';

export const UnifiedReportHead = NamedFC('UnifiedReportHead', () => {
    // tslint:disable: react-no-dangerous-html
    return (
        <head>
            <meta charSet="UTF-8" />
            <title>{androidAppTitle} - Report</title>
            <style dangerouslySetInnerHTML={{ __html: reportStyles.styleSheet }} />
            <style dangerouslySetInnerHTML={{ __html: bundledStyles.styleSheet }} />
        </head>
    );
});
