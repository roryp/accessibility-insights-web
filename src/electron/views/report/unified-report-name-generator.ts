// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ReportNameGenerator } from 'reports/report-name-generator';
import { ReportNameGeneratorBuilder } from 'reports/report-name-generator-builder';

export class UnifiedReportNameGenerator implements ReportNameGenerator {
    constructor(
        private readonly reportNameGeneratorBuilder: ReportNameGeneratorBuilder = new ReportNameGeneratorBuilder(),
    ) {}

    public generateName(baseName: string, scanDate: Date, pageTitle: string): string {
        return (
            baseName +
            '_' +
            this.reportNameGeneratorBuilder.getDateSegment(scanDate) +
            '_' +
            this.reportNameGeneratorBuilder.getTimeSegment(scanDate) +
            '_' +
            this.reportNameGeneratorBuilder.getTitleSegment(pageTitle) +
            '.html'
        );
    }
}
