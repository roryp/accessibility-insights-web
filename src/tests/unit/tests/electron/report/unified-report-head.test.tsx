// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { UnifiedReportHead } from 'electron/views/report/unified-report-head';
import { shallow } from 'enzyme';
import * as React from 'react';

describe('UnifiedReportHead', () => {
    it('renders', () => {
        const wrapper = shallow(<UnifiedReportHead />);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
