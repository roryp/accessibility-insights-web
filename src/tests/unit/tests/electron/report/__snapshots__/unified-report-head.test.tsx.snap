// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnifiedReportHead renders 1`] = `
<head>
  <meta
    charSet="UTF-8"
  />
  <title>
    Accessibility Insights for Android
     - Report
  </title>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "<<CSS:../reports/automated-checks-report.css>>",
      }
    }
  />
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "<<CSS:renderer.css>>",
      }
    }
  />
</head>
`;
